sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -15,12 +15,25 @@
 
 import re
 from collections import namedtuple
+from ._metadata import TypeMeta
 
 
 # TODO: Move this to a separate class
 # For now, this identifies a condition with only "==" operator supported.
 ConditionOperator = namedtuple('ConditionOperator', 'operator operand1 operand2')
 
+def _match_serialized_pipelineparam(payload: str):
```

Comment:
It's better to have a method called parse_pipelineparams(payload: str) > list(PipelineParam) to extract params. Any place that you just want to get the match but not the PipelineParam?

Reply:
Yes, in the compiler, it needs to resolve the dependencies, the unsanitized values are checked. 
Check out the compiler/compiler.py Line 222.

Reply:
sg, let's keep it.